src/main/java/uk/gov/justice/digital/delius/service/TierService.java

```diff
@@ -23,29 +27,50 @@
 public class TierService {
 
     private final ManagementTierRepository managementTierRepository;
-    private final StandardReferenceRepository standardReferenceRepository;
     private final TelemetryClient telemetryClient;
     private final OffenderRepository offenderRepository;
+    private final ReferenceDataService referenceDataService;
+    private final ContactService contactService;
+    private final StaffRepository staffRepository;
+    private final TeamRepository teamRepository;
 
 
     @Transactional
     public void updateTier(String crn, String tier) {
-        Map<String, String> telemetryProperties = Map.of("crn", crn, "tier", tier);
-        final Long offenderId = getOffender(crn, telemetryProperties);
+        final var tierWithUPrefix = tierWithUPrefix(tier);
+        final var telemetryProperties = Map.of("crn", crn, "tier", tierWithUPrefix);
+        final var offender = getOffender(crn, telemetryProperties);
 
-        writeTierUpdate(tier, telemetryProperties, offenderId);
+        final var offenderId = offender.getOffenderId();
+        final var changeReason = referenceDataService.getAtsTierChangeReason().orElseThrow(() -> logAndThrow(telemetryProperties, "TierUpdateFailureTierChangeReasonNotFound", "Tier change reason ATS not found"));
+
+        writeTierUpdate(tierWithUPrefix, telemetryProperties, offenderId, changeReason);
+
+        OffenderManager offenderManager = offender.getActiveCommunityOffenderManager().orElseThrow(() ->  logAndThrow(telemetryProperties, "TierUpdateFailureActiveCommunityOffenderManagerNotFound", String.format("Could not find active community manager for crn %s", crn)));
+        Staff staff;
+        Team team;
+        try {
+            String areaCode = offenderManager.getProbationArea().getCode();
```

Comment:
You could clean this up a bit by moving this logic to the StaffService or TeamService

staffService.findUTSStaff()  or something - or whatever UTS stands for

Reply:
I did have a quick look at this - I was a bit put off by the absence of simple finders on those services. This code really doesn't belong in this application - I'm mindful that it should be using the delius API really,  so I'm trying to keep it fairly self-contained by using existing methods where possible. 

It is a horrible mess - it would be a bit better if the fallback code were unnecessary.

Reply:
Sure